### PR TITLE
Make fee denom and amount configurable

### PR DESCRIPTION
**Pull request by andynog**
Closes: #754 

## Description

Implemented changes to allow `fee_denom` and `fee_amount` to be specified in the `config.toml` file. This change is required to allow Hermes to work with any Cosmos chain.
______

For contributor use:

- [ ] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [X] Updated relevant documentation (`docs/`) and code comments.
- [X] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: relayer/src/chain/cosmos.rs [hunk 1]
```diff
@@ -243,6 +238,18 @@ impl CosmosSdkChain {
         self.config.gas.unwrap_or(DEFAULT_MAX_GAS)
     }
 
+    fn fee(&self) -> Coin {
+        let amount = self
+            .config
+            .clone()
+            .fee_amount
+            .unwrap_or(DEFAULT_GAS_FEE_AMOUNT);
+        return Coin {
```
Comment (romac) [inline relayer/src/chain/cosmos.rs:247]:
```suggestion

        Coin {
```

File: relayer/src/chain/cosmos.rs [hunk 2]
```diff
@@ -243,6 +238,18 @@ impl CosmosSdkChain {
...
+            denom: self.config.clone().fee_denom,
```
Comment (romac) [inline relayer/src/chain/cosmos.rs:248]:
```suggestion
            denom: self.config.fee_denom.clone(),
```

File: relayer/src/chain/cosmos.rs [hunk 3]
```diff
@@ -243,6 +238,18 @@ impl CosmosSdkChain {
...
+            amount: (amount).to_string(),
```
Comment (romac) [inline relayer/src/chain/cosmos.rs:249]:
```suggestion
            amount: amount.to_string(),
```

**Review by romac — CHANGES_REQUESTED** [review]

**Comment by andynog** [comment]
Thanks

File: relayer/src/chain/cosmos.rs [hunk 4]
```diff
@@ -243,6 +238,19 @@ impl CosmosSdkChain {
...
+        let amount = self
+            .config
+            .clone()
+            .fee_amount
+            .unwrap_or(DEFAULT_GAS_FEE_AMOUNT);
+
+        Coin {
+            denom: self.config.fee_denom.clone(),
+            amount: amount.to_string(),
+        };
```
Comment (andynog) [inline relayer/src/chain/cosmos.rs:251]:
```suggestion
        }
```

**Review by andynog — COMMENTED** [review]
Fix to prevent e2e failure

**Review by romac — APPROVED** [review]

**Review by romac — CHANGES_REQUESTED** [review]
CI is failing in the end-to-end test. Looks like a config file is missing the `fee_denom` option.

**Review by adizere — APPROVED** [review]

**Review by romac — APPROVED** [review]

**Pull request merged**